### ISSUE TITLE
[Bug]: Images are not displayed if they have long file/folder names in Windows

### ISSUE THREAD

**Issue by ollm** [electron/electron#42634]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

31.0.2

### What operating system are you using?

Windows

### Operating System Version

Windows 11 Pro 23H2 22631.3737

### What arch are you using?

x64

### Expected Behavior

The image should display correctly without the need to do anything with the path.

### Actual Behavior

Currently, in some cases the images with long file names or in folders with also long names are not displayed correctly.

At the moment it has only happened to me if the image is inside a folder with a long name in the `Tmp` directory, but I don't know if it happens in other situations.

Something that I have noticed is that when it happens, if I click on the file information in File Explorer, it has the shortened path, and this path works in electron, but not the completed path, however, with `fs.existsSync`, for example, both work.

Testcase Screenshot:

![Captura de pantalla 2024-06-24 082548](https://github.com/electron/electron/assets/18549191/fb82ea78-4be3-4f96-9cc0-c1c9c1742bcb)

### Testcase Gist URL

https://github.com/ollm/electron-windows-path-error

**Comment by mlaurencin** [comment]
@ollm can you confirm whether or not you have long paths enabled? See [here](https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=registry#enable-long-paths-in-windows-10-version-1607-and-later) for more information.

I am adding the `blocked/need-info` label for now. If you are unable to respond within 10 days, the issue will be closed.

**Comment by ollm** [comment]
Thanks for the reply.

I'm not sure if I have long paths enabled, I have `LongPathsEnabled` set to 1 but `Enable Win32 long paths` is Not Configured, also the File Explorer doesn't let me create files with names longer than 221 characters and the testcase [file](https://github.com/ollm/electron-windows-path-error/blob/main/www/image/image-long-filename00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000.png) only has 154 characters, and the [folder](https://github.com/ollm/electron-windows-path-error/blob/e195991ae2569b2a9ffe606e1ffbc48e44f1005f/www/js/app.js#L29C43-L29C191) has 148 characters.

Here are the screenshots for reference:

![Captura de pantalla (192)](https://github.com/electron/electron/assets/18549191/660d9a29-5f61-47d1-8442-1f1d409cce5a)

![Captura de pantalla (193)](https://github.com/electron/electron/assets/18549191/f26d468f-29ec-43a5-8ff7-8acc16edefa5)

**Comment by mlaurencin** [comment]
If you followed the instructions in the document, you should have it enabled properly. 

From looking at your test, it appears that your first attempt opening it from the app folder has a path length around 247 characters, while your second attempt opening it from tmp is over 300 characters. If you increase the length of the first attempt to over 260, are you still able to open and display the image?

**Comment by mlaurencin** [comment]
Long paths on Windows can be a bit finicky. If following the document did not help, then I would suggest heading over to [the Electron Discord server](https://discord.com/invite/electron) and asking what people have done to work around this when testing locally. 

Once you go to package your app, you should be able to include `longPathAware` in your manifest. This is not something included by default.

As this definitely looks like a long path problem now, I am going to go ahead and close out this issue. If the problem still persists however after confirming long paths are properly enabled and the Discord is not able to help, feel free to open another issue. I hope all this info helps!

**Comment by ollm** [comment]
Thanks for your help! I'll try to get long paths working in Windows, I will update this once I find a way.

**Comment by ollm** [comment]
I've been looking and it seems that the `longPathAware` option only works for appx build, In the end I decided to generate the short path if the path is equal or greater than 260 characters (Windows [8.3 short filename](https://en.wikipedia.org/wiki/8.3_filename)), so it should solve this issue in most cases (If shorts paths has not been manually disabled in NTFS).

For those who have the same issue, I have created an [npm package](https://www.npmjs.com/package/short-windows-path) to generate/get the short version of a path.
[short-windows-path](https://github.com/ollm/short-windows-path)

I also found this other one, but personally I get errors with long paths.
[short-filename](https://github.com/particle-iot/short-filename)
